demo/src/App.js

```diff
@@ -1,233 +1,182 @@
+        [field]: dimIndex
+      });
+      dispatch({
+        type,
+        index,
+        value: { name: values[dimIndex], selection: serialized }
+      });
+    } else {
+      dispatch({ type, index, value });
+    }
   };
 
-  const toggleOverview = () => {
-    setOverview(prevOverviewOn => {
-      return !prevOverviewOn;
+  const handleChannelAdd = () => {
+    const [channelDim] = sources[sourceName].dimensions;
+    dispatch({
+      type: 'ADD_CHANNEL',
+      value: { name: channelDim.values[0], selection: [0, 0, 0] }
     });
   };
 
-  const sourceButtons = Object.keys(sources).map(name => {
+  const { initialViewState, isPyramid, dimensions } = sources[sourceName];
+  const { names, colors, sliders, isOn, ids, selections } = channels;
+  const channelControllers = ids.map((id, i) => {
     return (
-      // only use isPublic on the deployment
-      // eslint-disable-next-line no-restricted-globals
-      (location.host === 'viv.vitessce.io' ? sources[name].isPublic : true) && (
-        <Button
-          variant="contained"
-          key={name}
-          disabled={name === sourceName}
-          onClick={() => setSourceName(name)}
-        >
-          {name}
-        </Button>
-      )
-    );
-  });
-
-  const sliders = sources[sourceName].channelNames.map((channel, i) => {
-    return (
-      <div key={`container-${channel}`}>
-        <p>{channel}</p>
-        <div style={{ width: '100%', display: 'flex', position: 'relative' }}>
-          <Checkbox
-            onChange={() => toggleChannel(i)}
-            checked={channelIsOn[i]}
-            style={{
-              color: `rgb(${
-                colormapOn ? COLORMAP_SLIDER_CHECKBOX_COLOR : colorValues[i]
-              })`,
-              '&$checked': {
-                color: `rgb(${
-                  colormapOn ? COLORMAP_SLIDER_CHECKBOX_COLOR : colorValues[i]
-                })`
-              }
-            }}
-          />
-          <Slider
-            value={sliderValues[i]}
-            onChange={(event, value) => handleSliderChange(i, value)}
-            valueLabelDisplay="auto"
-            getAriaLabel={() => channel}
-            min={MIN_SLIDER_VALUE}
-            max={MAX_SLIDER_VALUE}
-            style={{
-              color: `rgb(${
-                colormapOn ? COLORMAP_SLIDER_CHECKBOX_COLOR : colorValues[i]
-              })`,
-              top: '7px'
-            }}
-            orientation="horizontal"
-          />
-        </div>
-      </div>
+      <Grid key={`channel-controller-${names[i]}-${id}`} item>
+        <ChannelController
+          name={names[i]}
+          channelOptions={dimensions[0].values}
+          disableOptions={sourceName === 'tiff'}
+          isOn={isOn[i]}
+          sliderValue={sliders[i]}
+          colorValue={colors[i]}
+          handleChange={(type, value) => handleControllerChange(i, type, value)}
+          colormapOn={colormap.length > 0}
+        />
+      </Grid>
```

Comment:
If there is possible an intention to wrap the whole controller up for export, would it make sense to separate this out into its own wrapper component?  If you think it would take too much time for this PR, that's ok. Maybe open an issue to investigate this?

Reply:
I'll have to think about this, and will open an issue.